syntax = "proto3";

package product;

service ProductService {
  rpc GetProducts (Empty) returns (ProductList) {}
  rpc GetProduct (ProductId) returns (Product) {}
  rpc CreateProduct (Product) returns (Product) {}
  rpc UpdateProduct (Product) returns (Product) {}
  rpc DeleteProduct (ProductId) returns (Empty) {}
}

message Empty {}

message ProductId {
  int32 id = 1;
}

message Product {
  int32 id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
  string image = 5;
}

message ProductList {
  repeated Product products = 1;
}
